test/Microsoft.AspNetCore.Mvc.Core.Test/Internal/ControllerArgumentBinderTests.cs

```diff
@@ -573,6 +574,122 @@ public class ControllerArgumentBinderTests
             Assert.Equal("Hello", controller.StringProperty);
         }
 
+        public static TheoryData BindingInfoData
+        {
+            get
+            {
+                var propertyFilterProvider = Mock.Of<IPropertyFilterProvider>();
+
+                var emptyBindingInfo = new BindingInfo();
+                var halfBindingInfo = new BindingInfo
+                {
+                    BinderModelName = "expected name",
+                    BinderType = typeof(Person),
+                };
+                var fullBindingInfo = new BindingInfo
+                {
+                    BinderModelName = "expected name",
+                    BinderType = typeof(Person),
+                    BindingSource = BindingSource.Services,
+                    PropertyFilterProvider = propertyFilterProvider,
+                };
+
+                var emptyBindingMetadata = new BindingMetadata();
+                var differentBindingMetadata = new BindingMetadata
+                {
+                    BinderModelName = "not the expected name",
+                    BinderType = typeof(TestController),
+                    BindingSource = BindingSource.ModelBinding,
+                    PropertyFilterProvider = Mock.Of<IPropertyFilterProvider>(),
+                };
+                var secondHalfBindingMetadata = new BindingMetadata
+                {
+                    BindingSource = BindingSource.Services,
+                    PropertyFilterProvider = propertyFilterProvider,
+                };
+                var fullBindingMetadata = new BindingMetadata
+                {
+                    BinderModelName = "expected name",
+                    BinderType = typeof(Person),
+                    BindingSource = BindingSource.Services,
+                    PropertyFilterProvider = propertyFilterProvider,
+                };
+
+                return new TheoryData<BindingInfo, BindingMetadata, BindingInfo>
```

Comment:
Leave a comment stating the parameters: `// parameterBindingInfo, bindingMetadata, expectedInfo`